storage/postgresql/schema/storage.sql

```diff
@@ -0,0 +1,224 @@
+-- PostgreSQL version of the tree schema.
+--
+-- Each statement must end with a semicolon, and there must be a blank line before the next statement.
+-- This will ensure that the testdbpgx tokenizer will handle semicolons in the PL/pgSQL functions correctly.
+
+-- ---------------------------------------------
+-- Tree stuff here
+-- ---------------------------------------------
+
+-- Tree parameters should not be changed after creation. Doing so can
+-- render the data in the tree unusable or inconsistent.
+CREATE TYPE TreeState AS ENUM ('ACTIVE', 'FROZEN', 'DRAINING');
+
+CREATE TYPE TreeType AS ENUM ('LOG', 'MAP', 'PREORDERED_LOG');
+
+
+CREATE TABLE IF NOT EXISTS Trees(
+  TreeId                BIGINT NOT NULL,
+  TreeState             TreeState NOT NULL,
+  TreeType              TreeType NOT NULL,
+  DisplayName           VARCHAR(20),
+  Description           VARCHAR(200),
+  CreateTimeMillis      BIGINT NOT NULL,
+  UpdateTimeMillis      BIGINT NOT NULL,
+  MaxRootDurationMillis BIGINT NOT NULL,
+  Deleted               BOOLEAN,
+  DeleteTimeMillis      BIGINT,
+  PRIMARY KEY(TreeId)
+);
+
+-- This table contains tree parameters that can be changed at runtime such as for
+-- administrative purposes.
+CREATE TABLE IF NOT EXISTS TreeControl(
+  TreeId                  BIGINT NOT NULL,
+  SigningEnabled          BOOLEAN NOT NULL,
+  SequencingEnabled       BOOLEAN NOT NULL,
+  SequenceIntervalSeconds INTEGER NOT NULL,
+  PRIMARY KEY(TreeId),
+  FOREIGN KEY(TreeId) REFERENCES Trees(TreeId) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS Subtree(
+  TreeId               BIGINT NOT NULL,
+  SubtreeId            BYTEA NOT NULL,
+  Nodes                BYTEA NOT NULL,
+  SubtreeRevision      INTEGER NOT NULL,
+  -- Key columns must be in ASC order in order to benefit from group-by/min-max
+  -- optimization in PostgreSQL.
+  CONSTRAINT Subtree_pk PRIMARY KEY (TreeId, SubtreeId, SubtreeRevision),
+  FOREIGN KEY(TreeId) REFERENCES Trees(TreeId) ON DELETE CASCADE,
+  CHECK (length(SubtreeId) <= 255)
+);
+
+-- The TreeRevisionIdx is used to enforce that there is only one STH at any
+-- tree revision
+CREATE TABLE IF NOT EXISTS TreeHead(
+  TreeId               BIGINT NOT NULL,
+  TreeHeadTimestamp    BIGINT,
+  TreeSize             BIGINT,
+  RootHash             BYTEA NOT NULL,
+  RootSignature        BYTEA NOT NULL,
+  TreeRevision         BIGINT,
```

Comment:
If my other comments about revisions were true, then this can be removed. And `TreeHeadRevisionIdx` can go.

Reply:
Addressed by https://github.com/google/trillian/pull/3644/commits/be9618289c8890e98a0aaf68c6f9bc20636343d6.